tests/network/expose.go

```diff
@@ -825,3 +803,27 @@ func generateHelloWorldServer(vmi *v1.VirtualMachineInstance, testPort int, prot
 		&expect.BExp{R: console.RetValue("0")},
 	}, 60)).To(Succeed())
 }
+
+func createAndWaitForJobToSucceed(jobFactory func(host, port string) *batchv1.Job, namespace, ip, port, viaMessage string) error {
+	By(fmt.Sprintf("Starting a job which tries to reach the VMI via the %s", viaMessage))
+	jobInstance := jobFactory(ip, port)
```

Comment:
This factory is executed only once. In a follow-up, please consider simplifying this further, by passing the Job object.

Passing functions by reference is cool, but when abused, it needlessly confuses the reader.